### ISSUE TITLE
Add default arguments to Node.AddComponent

### ISSUE THREAD

**Issue by JoshEngebretson** [AtomicGameEngine/AtomicGameEngine#354]
0 for id, replicated for mode

https://github.com/AtomicGameEngine/AtomicGameEngine/blob/master/Source/Atomic/Scene/Node.h#L548

**Comment by Alan-FGR** [comment]
I don't think that line is correct anymore. It's not the ctor, I wonder if this is relevant, I don't see where default ID would make sense it certainly doesn't make any to add default for setters.

**Comment by mattbenic** [comment]
https://github.com/AtomicGameEngine/AtomicGameEngine/blob/73e2d0e1caadd7d384996d7226d7d63547e7727a/Source/Atomic/Scene/Node.h#L612

**Comment by Alan-FGR** [comment]
Oops, totally misread the title :facepalm:
